src/services/analytics.ts

```diff
@@ -0,0 +1,50 @@
+import { reportInteraction } from '@grafana/runtime';
+import pluginJson from '../plugin.json';
+
+// Helper function to create a unique interaction name for analytics
+// TODO: Type action parameter
+const createInteractionName = (page: UserEventPagesType, action: string) => {
+  return `grafana_${pluginJson.id}_${page}_${action}`;
+};
+
+// Runs reportInteraction with a standardized interaction name
+// TODO: Type action parameter
```

Comment:
Wha's this todo for?

Reply:
Oh, I would like to type `action` parameter, so only `USER_EVENTS_ACTIONS[page]...` gets accepted. But I wasn't able to make it work and it is not a priority. I'll update it to make it more clear - what I meant.